common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -596,6 +597,40 @@ private void validateAuthScheme(@NonNull final MicrosoftStsTokenRequest request,
         }
     }
 
+    /**
+     * Validates that the token response contains an access token, id_token and refresh token
+     *
+     * @param response The idp response.
+     * @throws ClientException
+     */
+    private void validateTokensAreInResponse(@NonNull final MicrosoftStsTokenResponse response)
+            throws ClientException {
+
+        String clientException = null;
+        String tokens = "";
+        String tokensMissingMessage = "missing required tokens of type: {0}";
+
+        if (StringUtil.isEmpty(response.getIdToken())) {
+            clientException = ClientException.TOKENS_MISSING;
+            tokens.concat("id_token");
+        }
+
+        if (StringUtil.isEmpty(response.getAccessToken())) {
+            clientException = ClientException.TOKENS_MISSING;
+            tokens.concat(" access_token");
+        }
+
+        if (StringUtil.isEmpty(response.getRefreshToken())) {
+            clientException = ClientException.TOKENS_MISSING;
+            tokens.concat(" refresh_token");
+        }
```

Comment:
probably don't need to keep repeat this for each token:

`clientException = ClientException.TOKENS_MISSING;`

could instead just go off of the string length of `tokens`

Reply:
and then the `clientException` variable should not be needed